app/Http/Controllers/ArticleCategoryController.php

```diff
@@ -12,54 +17,58 @@ class ArticleCategoryController extends Controller
      */
     public function index()
     {
-        //
+        $query=ArticleCategory::Query();
+        $request=Request();
+        if($search=$request->Catergory){
+            $query->where("articleCategoryName","LIKE","%$search%");
+        }
+        $Categories=$query->get();
+        return view("Admin.article.allCategory",compact('Categories'));
     }
 
     /**
      * Show the form for creating a new resource.
      */
     public function create()
     {
-        //
+        return view("Admin.article.addCategory");
     }
 
     /**
      * Store a newly created resource in storage.
      */
-    public function store(Request $request)
-    {
-        //
-    }
-
-    /**
-     * Display the specified resource.
-     */
-    public function show(ArticleCategory $articleCategory)
+    public function store(CategoryArticleRequest $request)
     {
-        //
+        $data["articleCategoryName"]=$request["articleCategoryName"];
+        $data['hasComments']=isset($request['hasComments']);
+        $data['hasSource']=isset($request['hasSource']);
+        $data['hasYoutubeLink']=isset($request['hasYoutubeLink']);
+        $data['hasAuthor']=isset($request['hasAuthor']);
```

Comment:
You have a little duplication note 
```php
        $data["articleCategoryName"]=$request["articleCategoryName"];
        $data['hasComments']=isset($request['hasComments']);
        $data['hasSource']=isset($request['hasSource']);
        $data['hasYoutubeLink']=isset($request['hasYoutubeLink']);
        $data['hasAuthor']=isset($request['hasAuthor']);
```
is used in both store and update, what about extracting it into another private method that prepare the data